contracts/eosio.bios/src/eosio.bios.cpp

```diff
@@ -54,4 +57,82 @@ void bios::reqactivated( const eosio::checksum256& feature_digest ) {
    check( is_feature_activated( feature_digest ), "protocol feature is not activated" );
 }
 
+void bios::syscurrency() {
+    require_auth(get_self());
+
+    // Set the system resource currency to oncoin.tmy
+    eosio::symbol system_resource_currency = eosio::symbol("ONO", 4);  
+
+    // Set the resource token to oncoin.tmy
+    name resource_token_contract = eosio::name("oncoin.tmy");
+}
+
+
+void bios::setramprice(double new_price) {
+   require_auth("gov.tmy"_n); // check authorization is gov.tmy
+
+   // check new_price is within bounds, not negative or too high
+   eosio::check(new_price >= 0, "New price must be non-negative");
+
+   resource_config_table resource_configs(get_self(), get_self().value);
+      abi_hash_table table(get_self(), get_self().value);
+
+   auto itr = resource_configs.find(get_self().value);
+
+   if (itr == resource_configs.end()) {
+      // If record not found, create a new one
+      resource_configs.emplace(get_self(), [&](auto& row) {
+         row.ram_price = new_price;
+         row.total_ram_available = 0;
+         row.total_ram_used = 0;
+      });
+    } else {
+      // Modify existing record
+      resource_configs.modify(itr, get_self(), [&](auto& row) {
+         row.ram_price = new_price;
+      });
+    }
+}
+
+void bios::buyram(eosio::name dao_owner, eosio::name app, eosio::asset quant) {
+    require_auth(app); // Check that the app has the necessary authorization
+
+   // Access the account table from id.tmy.hpp
+   idtmy::id::account_types_table account_types("id.tmy"_n, "id.tmy"_n.value);
+   // Check the account type of the app
+   auto itr = account_types.find(app.value);
+   eosio::check(itr != account_types.end(), "Could not find account");
+   eosio::check(itr->account_type == idtmy::enum_account_type::App, "Only apps can buy and sell RAM");
+
+   // Check that the RAM is being purchased with the correct token
+   eosio::check(quant.symbol == core_symbol(), "must buy ram with core token");
+
```

Comment:
do we also need to check decimal place?

Reply:
yess i think otherwise it will throw mismatch error